decidim/decidim
# Enforce password validation rules on 'Forgot your password?' form

#### :tophat: What? Why?

Password validation rules didn't get enforced when a registered user reset their password through the "Forgot your password?" form (from Devise).

This PR fixes that. 

~~I guess we could remove the validation from the forms:~~

https://github.com/decidim/decidim/blob/05b988328ca9195462e2fb6ee9fbed854fb6c2ca/decidim-core/app/forms/decidim/registration_form.rb#L21
https://github.com/decidim/decidim/blob/05b988328ca9195462e2fb6ee9fbed854fb6c2ca/decidim-core/app/forms/decidim/account_form.rb#L28

~~But before changing that, I want to see the specs if it fails some flows that don't need passwords (like Invitations)~~

(Never mind, we need that so the Forms validations work correctly. If not it gives an error without any detail on what exactly is the error) 

#### :pushpin: Related Issues

- Fixes #9085 

#### Testing

1. Do the "Forgot your password?" flow until the password and confirmation screen
2. Fill a password that doesn't follow the rules, for instance "1234567890"
4. Save it 
5. See that it gives you a validation error 

:hearts: Thank you!
